src/engine/racer.rs

```diff
@@ -153,9 +142,6 @@ pub fn collapse_whitespace(text: &str) -> String {
   Regex::new(r"\s+").unwrap().replace_all(text, " ")
 }
 
-unsafe impl<'a> Sync for Racer<'a> {}
-unsafe impl<'a> Send for Racer<'a> {}
```

Comment:
Thanks for documenting why they are ok here. v2 of the Racer API won't expose IndexedSource at all and those impls will be able to go away.